notebooks/aws-emr/README.md

```diff
@@ -0,0 +1,74 @@
+## Running notebooks on AWS EMR 
+
+If you already have a AWS EMR account, you can run the example notebooks on an EMR cluster, as follows:
+- Install the [AWS CLI](https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-cli.html).
+- Initialize the CLI via `aws configure`. You may need to create access keys by following [Authenticating using IAM user credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-authentication-user.html). You can find your default region name (e.g. Ohio) on the right of the top navigation bar. Clicking the region name will show the region code (e.g. us-east-2 for Ohio). 
+  ```
+  aws configure
+  AWS Access Key ID [None]: <your_access_key>
+  AWS Secret Access Key [None]: <your_secret_access_key>
+  Default region name [None]: <region-code>
+  Default output format [None]: <json>
+  ```
+- Create an S3 bucket if you don't already have one.
+  ```
+  export S3_BUCKET=<your_gcs_bucket_name>
+  aws s3 mb s3://${S3_BUCKET}
+  ```
+- Create a zip file for the `spark-rapids-ml` package.
+  ```
+  cd spark-rapids-ml/python/src
+  zip -r spark_rapids_ml.zip spark_rapids_ml
+  ```
+- Upload the zip file and the initialization script to S3.
+  ```
+  aws s3 cp spark_rapids_ml.zip s3://${S3_BUCKET}/spark_rapids_ml.zip
+  cd ../../notebooks/aws-emr
+  aws s3 cp init-bootstrap-action.sh s3://${S3_BUCKET}/init-bootstrap-action.sh
+  ```
+- Print out available subnets in CLI then pick a SubnetId (e.g. subnet-0744566f of AvailabilityZone us-east-2a).
+
+  ```
+  aws ec2 describe-subnets
+  export SUBNET_ID=<your_SubnetId>
+  ```
+
+- Create a cluster with at least two single-gpu TASK workers. You will obtain a ClusterId in terminal.  
+
+  ```
+
+  export CLUSTER_NAME="spark_rapids_ml"
+  export CUR_DIR=$(pwd)
+
+  aws emr create-cluster \
+  --name ${CLUSTER_NAME} \
+  --release-label emr-6.10.0 \
+  --applications Name=Hadoop Name=Livy Name=Spark Name=JupyterEnterpriseGateway \
+  --service-role EMR_DefaultRole \
+  --ec2-attributes SubnetId=${SUBNET_ID},InstanceProfile=EMR_EC2_DefaultRole \
+  --instance-groups InstanceGroupType=MASTER,InstanceCount=1,InstanceType=m4.4xlarge \
+                    InstanceGroupType=CORE,InstanceCount=1,InstanceType=g4dn.2xlarge \
+                    InstanceGroupType=TASK,InstanceCount=1,InstanceType=g4dn.2xlarge \
+                    InstanceGroupType=TASK,InstanceCount=1,InstanceType=g4dn.2xlarge \
+  --configurations file://${CUR_DIR}/init-configurations.json \
+  --bootstrap-actions Name='Spark Rapids ML Bootstrap action',Path=s3://${S3_BUCKET}/init-bootstrap-action.sh
+  ```
```

Comment:
Might be nice to explain how MASTER, CORE, TASK map to the various spark nodes (in case folks want to adjust these in the future).  For instance, I'm wondering why we need three GPU nodes if the driver wasn't using GPU?

Reply:
Good suggestion. Just added a few words to explain why three GPU instances to get two workers.

Reply:
So EMR is _randomly_ picking either a CORE or a TASK node to run JupyterLab?  I would assume that TASK nodes would be reserved for the executors, so JupyerLab would run on CORE (which could theoretically be a CPU machine).

Reply:
Yeah, printed out the node can be CORE or TASK. There could be a way to specifically assign the node.

Reply:
CORE nodes run hdfs while TASK nodes do not.   That seems to be the main distinction and it doesn't constrain where the driver goes.   I guess we can force the driver to land on a cpu only node by giving that node more memory than the other nodes and setting the driver memory such that yarn must schedule it to that node.   The executors can then land on the gpu nodes based on the gpu resource requirements specified for executors.

Reply:
Ideally yarn should be 'smart' and not schedule a driver that doesn't request gpu resource to a gpu node when adequate resources are available on a cpu node.

Reply:
(1) The cost for large memory can be higher than small memory + GPU. AWS pricing shows m4.4xlarge (64G) costs 0.8/h, and g4dn.2xlarge (32G + GPU) costs  0.75/h. 

(2) Yeah, not sure if this favorable feature of selecting CPU/GPU nodes is supported. 

(3) Tried other option to avoid GPU driver like node-label https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-master-core-task-nodes.html. Did not get it to work.